salt/modules/dockerng.py

```diff
@@ -504,10 +504,14 @@ def __virtual__():
                 log.warning(
                     'Insufficient Docker version for dockerng (required: '
                     '{0}, installed: {1})'.format(
-                        '.'.join(docker_versioninfo),
-                        '.'.join(MIN_DOCKER)
-                    )
-                )
+                        '.'.join(map(str, MIN_DOCKER)),
+                        '.'.join(map(str, docker_versioninfo))))
+        else:
+            log.warning(
```

Comment:
@blueyed, rather than logging a warning here, which will add this log statement every time the loader tries to load this module even for users who don't use it, it is recommended to return the reason along with the return.
```python
return (False, 'Insufficient docker-py version.  Expected cool-version, got lame-version.')
```